src/main/io/gps.c

```diff
@@ -345,6 +373,76 @@ void gpsInitUblox(void)
     }
 }
 
+void gpsInitMTK(void)
+{
+    uint32_t now;
+
+    // Wait until GPS transmit buffer is empty
+    if (!isSerialTransmitBufferEmpty(gpsPort))
+        return;
+
+
+    switch (gpsData.state) {
+        case GPS_INITIALIZING:
+            now = millis();
+            if (now - gpsData.state_ts < GPS_BAUDRATE_CHANGE_DELAY)
+                return;
+
+            if (gpsData.state_position < GPS_INIT_ENTRIES) {
+                // try different speed to INIT
+                baudRate_e newBaudRateIndex = gpsInitData[gpsData.state_position].baudrateIndex;
+
+                gpsData.state_ts = now;
+
+                if (lookupBaudRateIndex(serialGetBaudRate(gpsPort)) != newBaudRateIndex) {
+                    // change the rate if needed and wait a little
+                    serialSetBaudRate(gpsPort, baudRates[newBaudRateIndex]);
+                    return;
+                }
+
+                // print our FIXED init string for the baudrate we want to be at
+                serialPrint(gpsPort, gpsInitData[gpsData.baudrateIndex].mtk);
+
+                gpsData.state_position++;
+            } else {
+                // we're now (hopefully) at the correct rate, next state will switch to it
+                gpsSetState(GPS_CHANGE_BAUD);
+            }
+            break;
+
+        case GPS_CHANGE_BAUD:
+            serialSetBaudRate(gpsPort, baudRates[gpsInitData[gpsData.baudrateIndex].baudrateIndex]);
+            gpsSetState(GPS_CONFIGURE);
+            break;
+
+        case GPS_CONFIGURE:
+// MTK config strings
+            if (gpsData.messageState == GPS_MESSAGE_STATE_IDLE) {
+                gpsData.messageState++;
+            }
+
+            if (gpsData.messageState == GPS_MESSAGE_STATE_INIT) {
+
+                if (gpsData.state_position < sizeof(MTK_Init)) {
```

Comment:
this will send trailing `\0` (C string terminator) too. Is it desired?
If `strlen()` is used, gcc should optimize it away, so it will have zero overhead ... Or just test for trailing zero.